File: Java/2566. Maximum Difference by Remapping a Digit.java
class Solution {
    public int minMaxDifference(int num) {
        String s = Integer.toString(num);
        String t = s;
        char first_none_nine = '9';

        for(char c: s.toCharArray()) {
            if (c != '9') {
                first_none_nine = c;
                break;
            }
        }

        s = s.replace(first_none_nine, '9');
        t = t.replace(t.charAt(0), '0');
        return Integer.parseInt(s) - Integer.parseInt(t);
    }
}

File: Python3/2566. Maximum Difference by Remapping a Digit.py
class Solution:
    def minMaxDifference(self, num: int) -> int:
        s = str(num)
        t = s
        first_none_nine = '9'

        for n in s:
            if n != '9':
                first_none_nine = n
                break
        
        max_num = s.replace(first_none_nine, '9')
        min_num = t.replace(t[0], '0')
        
        return int(max_num) - int(min_num)